nibabel/brikhead.py

```diff
@@ -0,0 +1,488 @@
+            raise AFNIError('Cannot create AFNIHeader from nothing.')
+        if type(header) == klass:
+            return header.copy()
+        raise AFNIError('Cannot create AFNIHeader from non-AFNIHeader.')
+
+    @classmethod
+    def from_fileobj(klass, fileobj):
+        info = parse_AFNI_header(fileobj)
+        return klass(info)
+
+    def copy(self):
+        return AFNIHeader(deepcopy(self.info))
+
+    def _calc_data_shape(self):
+        """Calculate the output shape of the image data
+
+        Returns length 3 tuple for 3D image, length 4 tuple for 4D.
+
+        Returns
+        -------
+        (x, y, z, t) : tuple of int
+        """
+        dset_rank = self.info['DATASET_RANK']
+        shape = tuple(self.info['DATASET_DIMENSIONS'][:dset_rank[0]])
+        n_vols = dset_rank[1]
+
+        return shape + (n_vols,) if n_vols > 1 else shape
+
+    def _calc_zooms(self):
+        """ Get image zooms from header data
+
+        Spatial axes are first three indices
+
+        Returns
+        -------
+        zooms : tuple
+        """
+        xyz_step = tuple(np.abs(self.info['DELTA']))
+        t_step = self.info.get('TAXIS_FLOATS', ())
+        if len(t_step) > 0:
+            t_step = (t_step[1],)
+
+        return xyz_step + t_step
+
+    def get_space(self):
+        """Returns space of dataset
+
+        Returns
+        -------
+        space : str
+        """
+        listed_space = self.info.get('TEMPLATE_SPACE', 0)
+        space = space_codes.space[listed_space]
+
+        return space
+
+    def get_affine(self):
+        """ Returns affine of dataset
+
+        Examples
+        --------
+        >>> header = AFNIHeader(parse_AFNI_header('example4d+orig.HEAD'))
+        >>> header.get_affine()
+        array([[ -3.    ,  -0.    ,  -0.    ,  49.5   ],
+               [ -0.    ,  -3.    ,  -0.    ,  82.312 ],
+               [  0.    ,   0.    ,   3.    , -52.3511],
+               [  0.    ,   0.    ,   0.    ,   1.    ]])
+        """
+        # AFNI default is RAI-/DICOM order (i.e., RAI are - axis)
+        # need to flip RA sign to align with nibabel RAS+ system
+        affine = np.asarray(self.info['IJK_TO_DICOM_REAL']).reshape(3, 4)
+        affine = np.row_stack((affine * [[-1], [-1], [1]],
+                               [0, 0, 0, 1]))
+
+        return affine
+
+    def get_data_scaling(self):
+        """ AFNI applies volume-specific data scaling
+
+        Examples
+        --------
+        >>> header = AFNIHeader(parse_AFNI_header('scaled+tlrc.HEAD'))
+        >>> header.get_data_scaling()
+        array([  3.88336300e-08])
+        """
+        floatfacs = self.info.get('BRICK_FLOAT_FACS', None)
+        if floatfacs is None or not np.any(floatfacs):
+            return None
+        scale = np.ones(self.info['DATASET_RANK'][1])
+        floatfacs = np.atleast_1d(floatfacs)
+        scale[floatfacs.nonzero()] = floatfacs[floatfacs.nonzero()]
+
+        return scale
+
+    def get_slope_inter(self):
```

Comment:
Because `AFNIArrayProxy` inherits from `ArrayProxy`, this needs to be included. `ArrayProxy` demands that the provided "`spec`" (i.e., `AFNIHeader` in this case) implement this (see: arrayproxy.py:140). I added the inheritance in response to @effigies suggestion, above.

Reply:
Maybe add comment to this effect?